### ISSUE TITLE
implementing length_regular with tensorflow discusstion

### ISSUE THREAD

**Issue by superhg2012** [xcmyz/FastSpeech#45]
Hi, I am implementing fastspeech with tensorflow, when rewrite length regular module,  I met some problem. tf graph is static graph, I can not get real tensor shape before feeding data as I define inputs with placeholder and shape=[None,None, None].

my code is :

`
    def len_regulator(self, phoneme_seqs, duration_seqs, alpha=1.0, max_mel_length=None):
       
        D = tf.keras.backend.round(tf.scalar_mul(alpha, duration_seqs))

        # grouping
        pho_splits = tf.split(phoneme_seqs, num_or_size_splits=phoneme_seqs.shape.as_list()[-1], axis=0)
        dur_splits = tf.split(D, num_or_size_splits=D.shape.as_list()[-1], axis=0)

        repeats = [tf.ones(tf.cast(r, tf.int32), dtype=tf.float32) for r in dur_splits]

        expanded = list()
        for i, j in zip(pho_splits, repeats):
            expanded.append(tf.multiply(i, j))
        expanded = tf.concat(expanded, axis=0)
        return expanded
`

anybody who can help? thanks in advance!!

**Comment by alokprasad** [comment]
@superhg2012 where you able to proceed further, solving above issue?

**Comment by superhg2012** [comment]
@alokprasad not yet,nealy give up

**Comment by xiaoyangnihao** [comment]
@superhg2012  Hi, do you solve this problem, I am stuck in this problem for several days when trying to implement LengthRegulator  in tensorflow2. Hoping for your replay soon.

**Comment by dathudeptrai** [comment]
@superhg2012 @xiaoyangnihao https://github.com/dathudeptrai/TensorflowTTS. I implemented FastSpeech here

**Comment by xiaoyangnihao** [comment]
> @superhg2012 @xiaoyangnihao https://github.com/dathudeptrai/TensorflowTTS. I implemented FastSpeech here

thanks for your nice work!

**Comment by superhg2012** [comment]
> @superhg2012 @xiaoyangnihao https://github.com/dathudeptrai/TensorflowTTS. I implemented FastSpeech here

great! thanks!!
